tests/integration/test_source_git.py

```diff
@@ -641,3 +641,52 @@ def test_srpm_patch_non_conseq_indices(
         "0004-Wei-bier-Summer-is-coming.patch",
     }
     assert sg_path.joinpath(DISTRO_DIR, "saaz.patch").read_text() == ""
+
+
+def test_add_patch_with_patch_id(
+    mock_remote_functionality_sourcegit, api_instance_source_git
+):
+    """check that patches with patch_id set are added to spec correctly"""
+    spec_dir = api_instance_source_git.up.absolute_specfile_dir
+    spec: Specfile = api_instance_source_git.up.specfile
+
+    # we want to add this patch to the spec
+    good_patch_name1 = "hello.patch"
+    # we need to create the patch file so that rebase-helper can find it and process it
+    good_patch_path1 = spec_dir.joinpath(good_patch_name1)
+    good_patch_path1.write_text("")
+    good_patch1 = PatchMetadata(
+        name=good_patch_name1,
+        path=good_patch_path1,
+        present_in_specfile=False,
+        patch_id=0,
```

Comment:
Nice improvement on the test. Would it be possible to parameterize this to have:

- starting with 0: (patch_id=0, no patch_id, patch_id=1)
- starting higher: (patch_id=100, no patch_id, patch_id=99)

I feel that having only this only one scenario some issue could hide behind the second expectation of patch_id == 1, b/c 1 is also the default patch_id.